core/src/main/java/hudson/model/Job.java

```diff
@@ -1591,27 +1565,18 @@ private Calendar getLastBuildTime() {
     public/* not synchronized. see renameTo() */void doDoRename(
             StaplerRequest req, StaplerResponse rsp) throws IOException,
             ServletException {
+        super.doDoRename2(req, rsp);
+    }
 
-        if (!hasPermission(CONFIGURE)) {
-            // rename is essentially delete followed by a create
-            checkPermission(CREATE);
-            checkPermission(DELETE);
-        }
-
-        String newName = req.getParameter("newName");
-        Jenkins.checkGoodName(newName);
-
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void checkRename(String newName) throws Failure {
+        Jenkins.get().getProjectNamingStrategy().checkName(newName);
```

Comment:
Should be pulled up to `AbstractItem.doCheckNewName`—it is not specific to `Job`.